src/deckgl-layers/3d-building-layer/3d-building-layer.js

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {CompositeLayer} from '@deck.gl/core';
+import DeckGLTileLayer from '@deck.gl/experimental-layers/dist/tile-layer/tile-layer';
+import {getTileData} from 'utils/3d-building-utils';
+import {_SolidPolygonLayer as SolidPolygonLayer} from '@deck.gl/layers';
+
+export default class ThreeDBuildingLayer extends CompositeLayer {
+  // this layer add its subLayers to the redux store, and push sample data
+  
+  renderSubLayers(props) {
+    return new SolidPolygonLayer({
+      ...props,
+      parameter: {
+        blendFunc: [
+          'SRC_ALPHA',
+          'ONE_MINUS_SRC_ALPHA',
+          'ONE',
+          'ONE_MINUS_SRC_ALPHA'
+        ],
+        blendEquation: ['FUNC_ADD', 'FUNC_ADD']
+      },
+      extruded: true,
+      opacity: 1,
+      filled: true,
+      getElevation: (feature) => feature.properties.height || 0,
+      getPolygon: (feature) => feature.coordinates,
+      getFillColor: this.getFillColor(),
+      lightSetting: {
+        ambientRatio: 0.2
+      }
+    });
+  }
+
+  getFillColor() {
```

Comment:
Should this logic be in the deckgl-layer? I feel like this should be a prop saved in the map-style reducer, and allow people to set it, or automatically determined by current map style color. Previously it was implemented like this:
[Props in map-style reducer](https://code.uberinternal.com/diffusion/VIVOY/browse/building-optimization/src/shared/reducers/map-style.js$43-47)     
[getBuildingColor](https://code.uberinternal.com/diffusion/VIVOY/browse/building-optimization/src/shared/reducers/map-style.js$92-101)

Reply:
I agree the color should be a prop if we want this layer to be a controlled component. How can users set it though?

Reply:
I believe the idea is to figure out the proper color based on map-style in the reducer and pass it to MapContainer.

Reply:
I've moved this function to the reducer, but it's still hardcoding color values. What's the logic of determining the color based on map style? I'm not sure what [this function](https://code.uberinternal.com/diffusion/VIVOY/browse/building-optimization/src/shared/reducers/map-style.js$92-101) is doing.

Reply:
the function is looking in the current map style json object stored in the reducer, and try to see if it has a 2D `building-fill` layer, if it does, it will use the color and bring up or down the brightness a little bit